tests/integration/test_encrypted_disk/test.py

```diff
@@ -580,3 +584,127 @@ def test_backup_restore(
+    assert node.query(select_query) == "(0,'data'),(1,'data')"
+
+    key1_arn, key1_encrypted_base64 = _create_aws_kms_key(key1)
+
+    # Exchange the new encrypted key through KMS with the original one
+    keys = f"""<no_sign_request>true</no_sign_request>
+            <aws_kms_endpoint>{cluster.local_kms_url}</aws_kms_endpoint>
+            <key_aws key_arn="{key1_arn}">{key1_encrypted_base64}</key_aws>"""
+
+    make_storage_policy_with_keys(
+        "encrypted_policy_multikeys", keys, check_system_storage_policies=True
+    )
+
+    select_query = "SELECT * FROM encrypted_test ORDER BY id FORMAT Values"
+    assert node.query(select_query) == "(0,'data'),(1,'data')"
+
+    # Ensure current_key_aws works
+    key2 = "secondsecondseco"
+    key2_arn, key2_encrypted_base64 = _create_aws_kms_key(key2)
+    keys = f"""<no_sign_request>true</no_sign_request>
+            <aws_kms_endpoint>{cluster.local_kms_url}</aws_kms_endpoint>
+            <key_aws key_arn="{key1_arn}">{key1_encrypted_base64}</key_aws>
+            <key_aws key_arn="{key2_arn}">{key2_encrypted_base64}</key_aws>
+            <current_key_aws>{key2_encrypted_base64}</current_key_aws>"""
+
+    make_storage_policy_with_keys(
+        "encrypted_policy_multikeys", keys, check_system_storage_policies=True
+    )
+
+    node.query("INSERT INTO encrypted_test VALUES (2,'data'),(3,'data')")
+
+    # Now "(0,'data'),(1,'data')" is encrypted with the first key and "(2,'data'),(3,'data')" is encrypted with the second key.
+    # All data are accessible.
+    assert node.query(select_query) == "(0,'data'),(1,'data'),(2,'data'),(3,'data')"
+
+    # Keys can be reordered
+    keys = f"""<no_sign_request>true</no_sign_request>
+            <aws_kms_endpoint>{cluster.local_kms_url}</aws_kms_endpoint>
+            <key_aws id="1" key_arn="{key2_arn}">{key2_encrypted_base64}</key_aws>
+            <key_aws id="0" key_arn="{key1_arn}">{key1_encrypted_base64}</key_aws>
+            <current_key_id>1</current_key_id>"""
+
+    make_storage_policy_with_keys(
+        "encrypted_policy_multikeys", keys, check_system_storage_policies=True
+    )
+
+    assert node.query(select_query) == "(0,'data'),(1,'data'),(2,'data'),(3,'data')"
+
+    # Try to replace the first key with something wrong, and check that "(0,'data'),(1,'data')" cannot be read.
+    keys = f"""<no_sign_request>true</no_sign_request>
+            <aws_kms_endpoint>{cluster.local_kms_url}</aws_kms_endpoint>
+            <key_aws id="1" key_arn="{key2_arn}">{key2_encrypted_base64}</key_aws>
+            <key id="0">wrongkey</key>
+            <current_key_id>1</current_key_id>"""
+    make_storage_policy_with_keys("encrypted_policy_multikeys", keys)
+
+    expected_error = "Not found an encryption key required to decipher"
+    assert expected_error in node.query_and_get_error(select_query)
+
+    # Detach the part encrypted with the wrong key and check that another part containing "(2,'data'),(3,'data')" still can be read.
```

Comment:
do we guarantee this table layout: two parts each of two rows? it of course doesn't affect the feature itself, but the test stability i guess.

Reply:
This test is just as the rest of the tests within this file: they all assume a part per insert.